src/h3/api/basic_int/__init__.py

```diff
@@ -564,30 +574,27 @@ def h3shape_to_cells_experimental(h3shape, res, flags=0):
     There is currently no guaranteed order of the output cells.
     """
 
-    if isinstance(flags, str):
-        try:
-            flags = ContainmentMode[flags]
-        except KeyError as e:
-            raise ValueError('Unrecognized flags: ' + flags) from e
-    if isinstance(flags, ContainmentMode):
```

Comment:
Personally I would prefer to keep the enum option in addition to the stringly-typed form.

Would typing pick up the available string options and offer them as suggestions? I don't see them in type annotations right now.

Reply:
I think typing in the future is the more natural way to handle this. We don't currently have types, but they could be added. We don't use them in the area/length functions, for example, to specify the output units.

> Would typing pick up the available string options and offer them as suggestions? I don't see them in type annotations right now.

Yes.

Reply:
And actually, if you're interested in adding support for IDE tooling like suggestions, I think we should do that in a separate issue or PR. I think that's outside the scope of this diff, as it affects the library as a whole.

And I'd definitely welcome that discussion, as I think it is needed!

Reply:
You know what? The function is experimental, so let's just try it and see how it works. I do still think we should separately take a look at typing in the library holistically.